src/utilities/oas-validator.ts

```diff
@@ -32,47 +50,45 @@ class OASValidator {
 
   validateParameters = async (
     operationId: string,
-    parameters: ParameterExamples,
+    parameterExamples: ParameterExamples,
   ): Promise<ValidationFailure[]> => {
-    const parameterSchema: {
-      [parameterName: string]: Parameter;
-    } = {};
     let failures: ValidationFailure[] = [];
 
     const operation = await this.schema.getOperation(operationId);
     if (!operation) {
       return [new InvalidOperationId(operationId)];
     }
-    const requiredParameters = operation.parameters
-      .filter((parameter) => parameter.required)
-      .map((parameter) => parameter.name);
-
-    const presentParameterNames = Object.keys(parameters);
-    const missingRequiredParameters = requiredParameters?.filter(
-      (parameterName) => !presentParameterNames.includes(parameterName),
+    const missingRequiredParametersError = this.checkMissingRequiredParameters(
+      operation,
+      parameterExamples,
     );
-
-    if (missingRequiredParameters.length > 0) {
-      failures = [
-        ...failures,
-        new MissingRequiredParameters(missingRequiredParameters),
-      ];
+    if (missingRequiredParametersError) {
+      failures = [...failures, missingRequiredParametersError];
     }
 
+    const operationParameters: OperationParameters = {};
     operation.parameters.forEach((parameter) => {
-      parameterSchema[parameter.name] = parameter;
+      operationParameters[parameter.name] = parameter;
     });
 
-    Object.entries(parameters).forEach(([key, value]) => {
-      if (Object.keys(parameterSchema).includes(key)) {
-        failures = [
-          ...failures,
-          ...OASValidator.validateObjectAgainstSchema(
-            value,
-            parameterSchema[key].schema,
-            ['parameters', key, 'example'],
-          ),
-        ];
+    Object.entries(parameterExamples).forEach(([name, example]) => {
```

Comment:
A foreach with an if and a sub if-elseif is a little complex for my simple brain. Can we break this up into different functions or simplify the cyclomatic complexity?

Reply:
I've shifted the body into a separate function, please let me know if this is better?